test/unit/template_importer_test.rb

```diff
@@ -142,5 +142,16 @@ def self.import!(name, text, _metadata)
       assert_equal File.read(get_template('ptable1.erb')), pt.layout
       assert_equal 'Debian', pt.os_family
     end
+
+    test 'purge! removes expected templates' do
+      FactoryGirl.create(:provisioning_template, :name => 'FooBar delete_me')
+      FactoryGirl.create(:provisioning_template, :name => 'FooBar delete_me_too')
+      FactoryGirl.create(:provisioning_template, :name => 'keep_me')
+      count = ProvisioningTemplate.all.size
+      @importer.purge!
+      refute ProvisioningTemplate.find_by_name('FooBar %').present?
+      assert ProvisioningTemplate.find_by_name('keep_me').present?
+      assert ProvisioningTemplate.all.size == count - 2
+    end # 'purge! removes expected template'
```

Comment:
You should probably add a test that shows the `negate` working, so that we don't break that function in later patches. Should be pretty easy, but do it as a new `test` block rather than more assertions in this one.